### ISSUE TITLE
IDEA: Hide elements that scroll out of view.

### ISSUE THREAD

**Issue by BeekeeperAlex** [matthieua/WOW#136]
In some cases I think it would be really cool to be able to tell wow to hide elements that scroll out of view. Maybe a separate class name like `wow-hide` or something like that. I have a good example on a page I maintain for a non-profit organization called Paw Project.

http://pawproject.org/faq

The FAQ page animates elements in as you scroll. It would be cool if as soon as the elements scroll out of view they hide themselves and then trigger again when you scroll back up. It would create a really neat effect.

What do you guys think? If you like the idea I may try to implement it.

**Comment by VerySoong** [comment]
:heavy_plus_sign: :one: It will be great to have this :metal:

Ideally is to have the animation options to hide/fade/slide out the elements in opposite directions when they are out of view. Otherwise, a simple fade out (like setting a default) for all wow animation is :thumbsup: too

**Comment by Nunnsy** [comment]
Great idea :+1: as animate.css already has the functions built in, just need a way of triggering them, perhaps with an offset of its own so the user can see it fade away?

**Comment by allanalexandre** [comment]
+ 1.. Good guys, I believe it would be great to have this implemented in reality, would act as the ' reset ' property of the plugin (scrollReveal.js).

< div date-sr = "enter bottom and scale up 20% over 2s reset" > Foo </div>

bringing the wow.js, could be more or less like this:

< div class = "wow animated fadeInX reset" > Foo </div> or [data-reset="true"]

is it possible to have this implemented?

**Comment by paulunstech** [comment]
This is something I've been trying to crack for some time, but I have not been able to find an answer, since I am not a programmer or an extensive coder... 

I have experience with Superscrollorama, http://johnpolacek.github.io/superscrollorama/, which is awesome, it uses Greensock animation, but it's very time consuming and heavy. WOW is just simple and easy... so.... PLEASE - how can this feature be added?!? I am sure it cannot be TOO difficult, for an expert coder!

If this was part of WOW, I'd be using it on every site I design. Period. Glad to see it is now a conversation!

**Comment by barrylachapelle** [comment]
I +1 this heavily and completely agree with @paulunstech I just spent a bit trying to removing classes with a callback. I think it might take a lot more than this though. WOW would need to have an offsettop param.

Mathieu this addition would blow WOW right up.

**Comment by sheadawson** [comment]
+1. That is all :D

**Comment by jamielovelace** [comment]
much +1, very yes

**Comment by titos2k** [comment]
+1

**Comment by abukwaik** [comment]
+1 too really cool idea!

**Comment by FredSarmento** [comment]
+1

**Comment by der-krueger** [comment]
+1

**Comment by yosemite-windows** [comment]
Yes an option to rewind any animation set by animate.css would be wonderful !

**Comment by RadianSmile** [comment]
+1

**Comment by atomgroomdesign** [comment]
I did this here, without Wow.js.  It would be cool to integrate something similar. 
https://github.com/atomgroomdesign/icon-exit-animations

**Comment by iamsmae** [comment]
I have been looking for a way to do this! I hope it gets added soon!

**Comment by JoshCarey** [comment]
I'll throw my hat in and also say that this would be an awesome feature.  I'm definitely no programmer, but I could probably spend way too much time getting this functionality to work with something like waypoints.js - https://github.com/imakewebthings/waypoints - or something similar.  Wow.js is just so clean and simple, I'd greatly prefer to use it everywhere for this type of stuff.  I'm definitely sad to see this still hasn't been figured out after all the time in the thread.

I think having the an explanation in the documentation say something to the tune of "if you want to reverse your animation when the element is scrolled out of view, add xxx-data-attribute= "opposite-animate.css-animation".

**Comment by kat2014** [comment]
I too am looking for ths type of solution with wow. Currently i am using classie.js but I have to manually determine and reset shrinkOn value off the page distance scrolled each time i add or remove elements to my html which is becoming impossible to manage. I woudl love to transition to using wow, as it seems to be a really nice and simple solution otherwise.

**Comment by redesigned** [comment]
+1 I'd even just settle for resetting the element when it leaves the page and re-revealing it when it re-enters the page.  Which should be fairly simple, right?

**Comment by jscontrust** [comment]
+1

**Comment by sahibvirk** [comment]
+1000

**Comment by graingert** [comment]
Pull requests accepted on https://github.com/graingert/wow the official https://wow.js.org source

**Comment by FakieSui** [comment]
+1 :-)
